ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v1/container/clientlibs/editor/js/editDialog.js

```diff
@@ -80,12 +80,25 @@
     }
 
     function handleFDMSubmitActionDialog(dialog) {
-        // Temporary Fix. Hide bindRefSelector Button
+        const formId = ns.ContentFrame.getDocument()[0].querySelector('[data-cmp-is="adaptiveFormContainer"]').id
+        function handleSubDialog(event) {
+            const bindRef = $(event.target).closest('.bindRefSelector').find('.bindRefTextField');
+            const options = {
+                formId: formId,
+                onSelect: function(selectedPath) {
+                    bindRef[0].value = selectedPath;
+                    return Promise.resolve();
+                }
+            };
+            ns.afUtils.generateSchemaTreeDialog(options);
+        }
         $(document).on("fdBindRef-contentLoaded", function (e) {
```

Comment:
bindRefSelector , bindRefTextField, bindRefSelectorButton  does not conforms to bem

fdBindRef-contentLoaded -> is it required? this looks interna.

Reply:
as it is internal, can we move this whole code to cq-guides as the submit sub dialog is internal so are its selectors.
`ns.afUtils.generateSchemaTreeDialog` will help us later in dor bind ref of edit dialog.

Reply:
Yes they are reqruied. The dialog listing the submit actions is rendered at server side https://github.com/adobe/aem-core-forms-components/blob/bbe999536762d960d6669033b2c2aa4a89b70e55/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/clientlibs/editor/utils/utils.js#L144

Reply:
IMO we should not access the elements that are not created using the open sourced code in the open sourced code. This will the code less transparent and unintuitive.